rotkehlchen/tests/unit/decoders/test_paraswap.py

```diff
@@ -0,0 +1,773 @@
+        balance=Balance(amount=FVal(swap_amount)),
+        location_label=user_address,
+        notes=f'Swap {swap_amount} SUSHI in paraswap',
+        counterparty=CPT_PARASWAP,
+        address=PARASWAP_AUGUSTUS_ROUTER,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=298,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_USDT,
+        balance=Balance(amount=FVal(received_amount)),
+        location_label=user_address,
+        notes=f'Receive {received_amount} USDT as the result of a paraswap swap',
+        counterparty=CPT_PARASWAP,
+        address=PARASWAP_AUGUSTUS_ROUTER,
+    )]
+    assert expected_events == events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0x711ee578860F6f621401A6148b09F69c1F8508B2']])
+def test_simple_swap_eth_fee(database, ethereum_inquirer, ethereum_accounts):
+    tx_hash = deserialize_evm_tx_hash(
+        '0xeeea20b39f157fe59fa4904fd4b62f8971188b53d05c6831e0ed67ee157e40c2',
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    user_address = ethereum_accounts[0]
+    timestamp = TimestampMS(1704806231000)
+    swap_amount = '3.020129914302378395'
+    received_amount = '2.990432283065311187'
+    gas_fees = '0.00680806307180382'
+    expected_events = [EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=0,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_ETH,
+        balance=Balance(amount=FVal(gas_fees)),
+        location_label=user_address,
+        notes=f'Burned {gas_fees} ETH for gas',
+        counterparty=CPT_GAS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=1,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=A_stETH,
+        balance=Balance(amount=FVal(swap_amount)),
+        location_label=user_address,
+        notes=f'Swap {swap_amount} stETH in paraswap',
+        counterparty=CPT_PARASWAP,
+        address=PARASWAP_AUGUSTUS_ROUTER,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=2,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_ETH,
+        balance=Balance(amount=FVal(received_amount)),
+        location_label=user_address,
+        notes=f'Receive {received_amount} ETH as the result of a paraswap swap',
+        counterparty=CPT_PARASWAP,
+        address=PARASWAP_AUGUSTUS_ROUTER,
+    )]  # it should also expect a fee event here, but it's not supported yet
```

Comment:
add it with a todo message. Create a note for this and link it here if you haven't